### PR TITLE
Task-38356: Add external label in tasks comments mention

### PR DESCRIPTION
**Pull request by ahmadmido**
This PR will display external label in tasks comments mention

### CODE REVIEW

**Comment by ahamdi** [comment]
This PR should be either on feature/maintenance or on your FT branch. You should not create PRs on develop directly

File: services/src/main/java/org/exoplatform/task/rest/TaskRestService.java [hunk 1]
```diff
@@ -609,18 +611,24 @@ public Response deleteComment(@ApiParam(value = "Comment id", required = true) @
...
 
   @GET
-  @Path("usersToMention/{query}")
+  @Path("usersToMention/{query}/{lang}")
   @RolesAllowed("users")
   @Produces(MediaType.APPLICATION_JSON)
   @ApiOperation(value = "Gets users to mention in comment", httpMethod = "GET", response = Response.class, notes = "This returns users to mention in comment")
   @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled") })
-  public Response findUsersToMention(@ApiParam(value = "Query", required = true) @PathParam("query") String query) throws Exception {
+  public Response findUsersToMention(@ApiParam(value = "Query", required = true) @PathParam("query") String query, @ApiParam(value = "Lang", required = true) @PathParam("lang") String lang) throws Exception {
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/task/rest/TaskRestService.java:619]: can you detect the locale directly from @Context HttpServletRequest request ?

File: services/src/main/java/org/exoplatform/task/util/TaskUtil.java [hunk 2]
```diff
@@ -1290,6 +1295,34 @@ private static TaskService getTaskService() {
     return container.getComponentInstanceOfType(TaskService.class);
   }
 
+  /**
+   * Check if the user is external
+   *
+   * @return true if the user is external
+   */
+  public static boolean isMemberOfExternalGroup(String userId) throws Exception{
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/task/util/TaskUtil.java]: to be done as the method Util.isExternal from social

**Review by azayati — CHANGES_REQUESTED** [review]
see review

File: services/src/main/java/org/exoplatform/task/service/TaskService.java [hunk 3]
```diff
@@ -153,4 +153,7 @@
     List<Object[]> countTaskStatusByProject(long projectId);
 
     public TasksList filterTasks(String query, long projectId, String keyword, List<Long> labels, TaskUtil.DUE dueDate, Priority priority, List<String> assignees, List<String> coworkers, List<String> watchers, Long labelId, Long statusId, Identity currIdentity, String dueCategory, String space_group_id , TimeZone userTimezone, boolean isShowCompleted, boolean advanceSearch, boolean noProjPermission, boolean noLblPermission, String orderBy, String groupBy, int offset, int limit) throws Exception;
+
+    public boolean isExternal(String userId) ;
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/task/service/TaskService.java:157]: Revert this modification

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: services/src/main/java/org/exoplatform/task/service/TaskService.java [hunk 4]
```diff
@@ -153,4 +153,7 @@
...
     }
+
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/task/service/TaskService.java:153]: format code

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — APPROVED** [review]
LGTM after QA validation

**Pull request merged**